pkg/controllers/provisioningv2/rke2/planner/controller.go

```diff
@@ -37,6 +38,14 @@ func Register(ctx context.Context, clients *wrangler.Context, planner *planner.P
 					Name:      clusterName,
 				}}, nil
 			}
+		} else if machine, ok := obj.(*capi.Machine); ok {
+			clusterName := machine.Labels[rke2.ClusterNameLabel]
+			if clusterName != "" {
+				return []relatedresource.Key{{
+					Namespace: machine.Namespace,
+					Name:      clusterName,
+				}}, nil
+			}
 		}
 		return nil, nil
 	}, clients.RKE.RKEControlPlane(), clients.Core.Secret())
```

Comment:
I believe we need to pass in a `clients.CAPI.Machine()` `ControllerWrapper` into this function in order for the above logic to work